document_merge_service/api/models.py

```diff
@@ -22,3 +25,31 @@ class Template(models.Model):
     engine = models.CharField(max_length=20, choices=ENGINE_CHOICES_TUPLE)
     group = models.CharField(max_length=255, db_index=True, blank=True, null=True)
     meta = models.JSONField(default=dict)
+
+
+@receiver(models.signals.post_delete, sender=Template)
+def auto_delete_file_on_delete(sender, instance, **kwargs):
+    """Delete template file from filesystem when `Template` object is deleted."""
+
+    if bool(instance.template):
+        if os.path.isfile(instance.template.path):
+            os.remove(instance.template.path)
+
+
+@receiver(models.signals.pre_save, sender=Template)
+def auto_delete_file_on_change(sender, instance, **kwargs):
+    """Delete old template file from filesystem when `Template` is given a new template file."""
+
+    if not instance.pk or not bool(instance.template):
+        return False
+
+    try:
+        old_file = Template.objects.get(pk=instance.pk).template
+    except Template.DoesNotExist:
+        return False
+
+    if bool(old_file):
```

Comment:
no need for explicit bool cast

Reply:
Removed the whole if since it's not used anymore. This was needed for the tests which had templates that were empty and the bool operator was neccessary to evaluate a `<FileField: None>` to `False`.

Reply:
done